packages/sdk/src/oracle/services/statemachines/fetchPastEvents.ts

```diff
@@ -41,6 +41,14 @@ export function Handlers(store: Store): GenericHandlers<Params, Memory> {
         await update.oracleEvents(chainId, currentStart, currentEnd);
         // reprocess all known events and create a table of requests from it
         await update.sortedRequests(chainId);
+
+        try {
+          // we can just try to update the current active request, we dont care if it fails, active request might not be set
+          await update.activeRequestFromEvents();
+        } catch (err) {
+          // do nothing
+        }
```

Comment:
@mrice32 i know you comment on this a lot, sorry to keep adding more of these, but i dont see a better way since in this case its ok to ignore the error, it should only throw if there is no active request ( user is on index page)

Reply:
If you expect it in the "happy path" then it shouldn't be an error :). Why can't we return a value? that indicates no active request? If all else fails, can we throw a _particular type of error_ that we can detect and ignore, so all other errors get passed through?

I think, in general, the architecture should allow for callers to either a) inspect enough to determine whether the method is callable in the current state or b) not error when called in that state and, even bettter, return a value that indicates a no-op so the caller gets feedback.

Do you think this is happening a lot because of the architecture of side-effect-only or write-only callers who can only induce actions but have no ability to see state or get feedback from methods via return values? I think if we want to use that sort of architecture, it might require no-throw code that doesn't produce errors, otherwise, you have to continually protect against edge cases by ignoring errors, burying real, problematic bugs and errors.

Reply:
i think you are touching on something fundamental here. you suggest a couple things, i think the one that makes the most sense to me is to check first if we can do something before trying it.  I've been using errors as a shorthand for doing that, with the assumption that if it throws an error it means i cant do it (without having to check prior to calling). As you point out theres room for truly unexpected errors which could be catastrophic that we are missing out on by doing this.  

i'll see about changing this to check for the conditions needed before calling this and see how that goes